akka-stream/src/main/scala/akka/stream/javadsl/Queue.scala

```diff
@@ -63,3 +63,14 @@ trait SinkQueue[T] {
    */
   def pull(): CompletionStage[Optional[T]]
 }
+
+/**
+ * This trait adds cancel support to [[SinkQueue]].
+ */
```

Comment:
Actually it's available for external usage - as it allow to stop upstream not waiting when it's finished by itself.
Mirroring the SourceQueueWithComplete